v0/destinations/drip/transform.js

```diff
@@ -0,0 +1,303 @@
+      customFields,
+      ["traits", "context.traits"],
+      IDENTIFY_EXCLUSION_FIELDS
+    );
+    if (!isEmptyObject(customFields)) {
+      payload.custom_fields = customFields;
+    }
+  }
+
+  payload = removeUndefinedAndNullValues(payload);
+  const finalpayload = {
+    subscribers: [payload]
+  };
+  const basicAuth = Buffer.from(Config.apiKey).toString("base64");
+  const response = defaultRequestConfig();
+  response.headers = {
+    Authorization: `Basic ${basicAuth}`,
+    "Content-Type": "application/json"
+  };
+  response.method = defaultPostRequestConfig.requestMethod;
+  const campaignId =
+    getDestinationExternalID(message, "dripCampaignId") || Config.campaignId;
+  if (campaignId && email) {
+    const check = await createUpdateUser(finalpayload, Config, basicAuth);
+    if (!check) {
+      throw new CustomError("Unable to create/update user.", 400);
+    }
+
+    let campaignPayload = constructPayload(message, campaignMapping);
+    campaignPayload.email = email;
+
+    campaignPayload = removeUndefinedAndNullValues(campaignPayload);
+    const finalCampaignPayload = {
+      subscribers: [campaignPayload]
+    };
+
+    response.endpoint = `${ENDPOINT}/v2/${Config.accountId}/campaigns/${campaignId}/subscribers`;
+    response.body.JSON = finalCampaignPayload;
+    return response;
+  }
+  response.endpoint = `${ENDPOINT}/v2/${Config.accountId}/subscribers`;
+  response.body.JSON = finalpayload;
+  return response;
+};
+
+const trackResponseBuilder = async (message, { Config }) => {
+  const id = getDestinationExternalID(message, "dripId");
+
+  let email = getFieldValueFromMessage(message, "email");
+  if (!isValidEmail(email)) {
+    email = null;
+    logger.error("Enter correct email format.");
+  }
+  let event = getValueFromMessage(message, "event");
+  if (event && ecomEvents.includes(event.trim().toLowerCase())) {
+    event = event.trim();
+    const personId = getDestinationExternalID(message, "person_id");
+    if (!personId && !email) {
+      throw new CustomError(
+        "Either person Id or email is required to make call.",
+        400
+      );
+    }
+    const payload = constructPayload(message, ecomMapping);
+    payload.email = email;
+    payload.person_id = personId;
+
+    if (payload.occurred_at && !isValidTimestamp(payload.occurred_at)) {
+      payload.occurred_at = null;
+      logger.error("Timestamp format must be ISO-8601.");
+    }
+    const productList = getValueFromMessage(message, "properties.product");
+    if (productList) {
+      const itemList = createList(productList);
+      if (itemList && itemList.length > 0) {
+        payload.items = itemList;
+      }
+    }
+    switch (event) {
+      case "order updated":
+        payload.action = "updated";
+        break;
+      case "order cancelled":
+        payload.action = "cancelled";
+        break;
+      case "order refunded":
+        payload.action = "refunded";
+        break;
+      case "order completed":
```

Comment:
for this one, instead of using switch case how about using a hashMap or simply an object. wdyt?